modules/simple-payments/paypal-express-checkout.js

```diff
@@ -119,37 +142,35 @@ var PaypalExpressCheckout = {
 					env: env
 				};
 
-				return paypal
-					.request
-					.post( PaypalExpressCheckout.getCreatePaymentEndpoint( blogId ), payload )
-					.then( function( paymentResponse ) {
-						return paymentResponse.id;
-					} )
-					.catch( function( paymentError) {
-						PaypalExpressCheckout.showError( 'Item temporarily unavailable', buttonDomId );
-					} );
+				return new paypal.Promise( function( resolve, reject ) {
+					jQuery.post( PaypalExpressCheckout.getCreatePaymentEndpoint( blogId ), payload )
+						.done( function( paymentResponse ) {
+							resolve( paymentResponse.id );
+						} )
+						.fail( function( paymentError ) {
+							var errorMessage = PaypalExpressCheckout.processErrorMessage( paymentError );
+							PaypalExpressCheckout.showError( errorMessage, buttonDomId );
+							reject( new Error( paymentError.responseJSON.code ) );
```

Comment:
We cannot get rid of those since Paypal promise system needs an error
![zrzut ekranu 2017-07-24 o 14 27 23](https://user-images.githubusercontent.com/3775068/28523454-c6788dc6-707c-11e7-9045-2402a8fe0d6f.png)